src/components/page/blocks/BlocksColumn.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Blocks from './index.jsx';
+
+function BlocksColumn({
+  getCol,
+  blocksGroups,
+  defaultLayout,
+  blockShared,
+  blockRows,
+}) {
+  const getLayout = layout => {
+    if (layout) {
+      const row = layout.row || defaultLayout.row;
+      const col = layout.col || defaultLayout.col;
+
+      return { row, col };
+    }
+
+    return defaultLayout;
+  };
+
+  const isInLayout = (layout, getRow) => {
+    const { row, col } = getLayout(layout);
+
+    return getCol === col && getRow === row;
+  };
+
+  const getRowBlockGroups = getRow =>
```

Comment:
this method could use some closer examination to see if it can be simplified. What it does is first uses a reducer on `blocksGroups` to remove the block groups that have no content (i.e. are `null`). 

Then with the block groups that do have content checks to see if they are within the current layout position. Finally since `filter()` will return an empty array for non-matching content blocks, it runs a second filter to remove those block groups that have empty arrays.

Reply:
What if we try address part of this where we're instantiating our `blocksGroups` array?  We can create a like `getBlocksGroups` method, and then have that method only return each block obj in the array if blocks of that type are on the page.  Pre-reducing so to speak.  And so if none of those obs were included, then `getRowBlockGroups` would never need to deal with null or empty array values for `blocks`, which I think would simplify this method almost to the point where it's just doing the like `isInLayout` filtering...maybe?  If we go this route, of course some careful, "ok, is there somewhere else that is depending on having those like 'empty' blockGroups".  And maybe there is, but we could probs weed those out without too much difficulty.